orderbook/src/main.rs

```diff
@@ -434,3 +436,84 @@ async fn check_database_connection(orderbook: &Orderbook) {
         .await
         .expect("failed to connect to database");
 }
+
+/// Parses a comma separated list of colon separated values representing fee factors for AppIds.
+fn parse_partner_fee_factor(s: &str) -> Result<HashMap<AppId, f64>> {
+    let mut res = HashMap::default();
+    for pair_str in s.split(',').into_iter().collect::<Vec<&str>>() {
```

Comment:
No need to `collect` in this line. We are only iterating so we don't need a Vec.